core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -282,7 +281,7 @@ class SparkContext(config: SparkConf) extends Logging {
    * @note As it will be reused in all Hadoop RDDs, it's better not to modify it unless you
    * plan to set some global configurations for all Hadoop RDDs.
    */
-  def hadoopConfiguration: Configuration = _hadoopConfiguration
+  def getHadoopConf: SparkHadoopConf = SparkHadoopConf.get()
```

Comment:
This is a public API, we can't directly remove this. Maybe just make this as deprecated?